pkgs/stdenv/generic/make-derivation.nix

```
@@ -577,14 +586,34 @@ lib.extendDerivation
        disallowedReferences = [ ];
        disallowedRequisites = [ ];
      });
-
-     inherit meta passthru overrideAttrs;
-   } //
+     inherit passthru;
+   }
+   // {
+     inherit meta overrideAttrs;
+   }
    # Pass through extra attributes that are not inputs, but
    # should be made available to Nix expressions using the
    # derivation (e.g., in assertions).
-   passthru)
-  (derivation (derivationArg // lib.optionalAttrs envIsExportable checkedEnv));
+   // passthru)
+  (
+      (
+        if attrs.__cleanAttrs or false
+        then builtins.intersectAttrs { version = null; } drvAttrs
+        else drvAttrs
+      )
+      // builtins.listToAttrs (map (outputName: {
+          name = outputName;
+          value = strict.${outputName};
+        }) outputs)
+      // {
+        type = "derivation";
+        inherit outputName outputs;
+        inherit (strict) drvPath;
+        inherit (drvAttrs) name;
+        ${if attrs.__cleanAttrs or false then null else "drvAttrs"} = drvAttrs;
```

```suggestion
        ${if attrs.__cleanAttrs or false then "internals" else "drvAttrs"} = drvAttrs;
```

Realistically, I think we'll want easier access than having to call a function to extract these bits. Having to go through a function means that the CLI becomes kind of useless, and same for tab completion in the repl.
"`internals`" does capture the intent of the attribute better: if you're going to use them, you're going to be affected by changes to internals, which are more likely than changes to the other attributes, which are implied to be a fairly stable interface. The name also opens up the possibility of exposing internals that aren't necessarily derivation arguments, but things that are currently in inaccessible `let` bindings.

---

If the content of `internals` und `drvAttrs` would be the same, maybe just keep `drvAttrs` as is? Changing the attr name introduces backward incompatibility.